ext/suse/pxp-agent.init

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+# pxp-agent        Init script for running the PCP Execution Protocol (PXP) agent daemon
+#
+# chkconfig: - 98 02
+#
+# description: Agent for the PCP Execution Protocol (PXP), based on
+#              the Puppet Communications Protocol (PCP).
+# processname: pxp-agent
+# config: /etc/sysconfig/pxp-agent
+
+### BEGIN INIT INFO
+# Provides: pxp-agent
+# Required-Start: $local_fs $remote_fs $network $syslog
+# Should-Start: pxp-agent
+# Required-Stop: $local_fs $remote_fs $network $syslog
+# Should-Stop: pxp-agent
+# Default-Start: 3 5
+# Default-Stop: 0 1 2 6
+# Short-Description: pxp-agent
+# Description: Agent daemon for the PCP Execution Protocol (PXP)
+### END INIT INFO
+
+# Shell functions sourced from /etc/rc.status:
+#      rc_check         check and set local and overall rc status
+#      rc_status        check and set local and overall rc status
+#      rc_status -v     ditto but be verbose in local rc status
+#      rc_status -v -r  ditto and clear the local rc status
+#      rc_failed        set local and overall rc status to failed
+#      rc_reset         clear local rc status (overall remains)
+#      rc_exit          exit appropriate to overall rc status
+[ -f /etc/rc.status ] && . /etc/rc.status
+[ -f /etc/sysconfig/pxp-agent ] && . /etc/sysconfig/pxp-agent
+
+exec=/opt/puppetlabs/puppet/bin/pxp-agent
+prog="pxp-agent"
+desc="PXP agent"
+
+lockfile=/var/lock/subsys/pxp-agent
+pidfile=/var/run/puppetlabs/pxp-agent.pid
+
+[ -x $exec ] || exit 5
+
+# First reset status of this service
+rc_reset
+
+# Return values acc. to LSB for all commands but status:
+# 0 - success
+# 1 - misc error
+# 2 - invalid or excess args
+# 3 - unimplemented feature (e.g. reload)
+# 4 - insufficient privilege
+# 5 - program not installed
+# 6 - program not configured
+#
+# Note that starting an already running service, stopping
+# or restarting a not-running service as well as the restart
+# with force-reload (in case signalling is not supported) are
+# considered a success.
+
+case "$1" in
+    start)
```

Comment:
the rpm script makes the pid dir if it doesn't exist already (for tmpfs systems), we probably want to do the same here, unless startproc makes the dir for us.